pkg/runtime/runtime.go

```diff
@@ -80,49 +80,45 @@ func (k *KubeadmRuntime) GetKubeadmConfig() ([]byte, error) {
 type Interface interface {
 	Init() error
 	Reset() error
-	JoinNodes(newNodesIPList []string) error
-	DeleteNodes(nodeIPList []string) error
-	JoinMasters(newMastersIPList []string) error
-	DeleteMasters(mastersIPList []string) error
+	ScaleNodes(newNodesIPList []string, deleteNodesIPList []string) error
```

Comment:
It may be more intuitive to refactor these methods into the following form.

```
// ScaleUp exec joining phase for cluster, add master role for these nodes. net.IP is the master node IP array.
	ScaleUp(newMasters, newWorkers []net.IP) error
	// ScaleDown exec deleting phase for deleting cluster master role nodes. net.IP is the master node IP array.
	ScaleDown(mastersToDelete, workersToDelete []net.IP) error
```